shell/app/common/components/pagination/index.tsx

```diff
@@ -0,0 +1,148 @@
+    onChange,
+  } = pagination;
+
+  const [goToVisible, setGoToVisible] = React.useState(false);
+
+  const paginationCenterRender = (
+    <Popover
+      content={<PaginationJump pagination={pagination} hiddenPopover={() => setGoToVisible(false)} />}
+      trigger="click"
+      getPopupContainer={(triggerNode) => triggerNode.parentElement as HTMLElement}
+      placement="top"
+      overlayClassName="pagination-jump"
+      visible={goToVisible}
+      onVisibleChange={setGoToVisible}
+    >
+      <div className="pagination-center bg-hover px-3 rounded cursor-pointer" onClick={() => setGoToVisible(true)}>
+        {total ? pagination.current : 0} / {(total && pageSize && Math.ceil(total / pageSize)) || 0}
+      </div>
+    </Popover>
+  );
+
+  const pageSizeMenu = (
+    <Menu>
+      {(pageSizeOptions || [])
+        .filter((item) => item)
+        .map((item: string | number) => {
+          return (
+            <Menu.Item key={item} onClick={() => onChange?.(1, +item)}>
+              <span className="fake-link mr-1">{i18n.t('{size} items / page', { size: item })}</span>
+            </Menu.Item>
+          );
+        })}
+    </Menu>
+  );
+
+  return (
+    <div className="erda-pagination flex justify-end items-center">
+      <div className="erda-pagination-total mr-2">共计 {total} 条</div>
+      <div className="erda-pagination-content inline-flex">
+        <div
+          className={`bg-hover p-2 leading-none ${current === 1 ? 'disabled' : ''}`}
+          onClick={() => current > 1 && onChange?.(current - 1)}
+        >
+          <ErdaIcon type="left" size={18} color="currentColor" />
+        </div>
+        {paginationCenterRender}
+        <div
+          className={`bg-hover p-2 leading-none ${current === Math.ceil(total / pageSize) ? 'disabled' : ''}`}
+          onClick={() => total && current < Math.ceil(total / pageSize) && onChange?.(current + 1)}
+        >
+          <ErdaIcon type="right" size={18} color="currentColor" />
+        </div>
+      </div>
+      {showSizeChanger && pageSizeOptions?.length && (
+        <Dropdown
+          trigger={['click']}
+          overlay={pageSizeMenu}
+          align={{ offset: [0, 5] }}
+          overlayStyle={{ minWidth: 120 }}
+          getPopupContainer={(triggerNode) => triggerNode.parentElement as HTMLElement}
+        >
+          <span className="bg-hover px-3 py-1 cursor-pointer">{i18n.t('{size} items / page', { size: pageSize })}</span>
+        </Dropdown>
+      )}
+    </div>
+  );
+};
+
+const PaginationJump = ({ pagination, hiddenPopover }: IPaginationJumpProps) => {
+  const { total = 0, pageSize = PAGINATION.pageSize, onChange } = pagination;
+  const [value, setValue] = React.useState('');
+
+  const handleChange = React.useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const { value: val } = e.target;
+      if (!isNaN(Number(val)) && +val <= Math.ceil(total / pageSize) && +val > 0 && !val.includes('.')) {
```

Comment:
refer to the logic of antd, if input number > last page no then jump to last page